src/remap/function/compact.rs

```diff
@@ -0,0 +1,406 @@
+                    ..Default::default()
+                },
+            ),
+            (
+                map!["key1": Value::from(1),
+                     "key3": Value::from(2)],
+                map!["key1": Value::from(1),
+                     "key2": Value::Array(vec![]),
+                     "key3": Value::from(2)],
+                CompactOptions {
+                    array: true,
+                    ..Default::default()
+                },
+            ),
+            (
+                map!["key1": Value::from(1),
+                     "key2": Value::Map(map!["key2": Value::from(3)]),
+                     "key3": Value::from(2),
+                ],
+                map![
+                    "key1": Value::from(1),
+                    "key2": Value::Map(map!["key1": Value::Null,
+                                            "key2": Value::from(3),
+                                            "key3": Value::Null]),
+                    "key3": Value::from(2),
+                ],
+                CompactOptions {
+                    recursive: true,
+                    null: true,
+                    ..Default::default()
+                },
+            ),
+            (
+                map!["key1": Value::from(1),
+                     "key3": Value::from(2),
+                ],
+                map![
+                    "key1": Value::from(1),
+                    "key2": Value::Map(map!["key1": Value::Null,]),
+                    "key3": Value::from(2),
+                ],
+                CompactOptions {
+                    recursive: true,
+                    null: true,
+                    map: true,
+                    ..Default::default()
+                },
+            ),
+            (
+                map!["key1": Value::from(1),
+                     "key2": Value::Array(vec![Value::from(2)]),
+                     "key3": Value::from(2),
+                ],
+                map![
+                    "key1": Value::from(1),
+                    "key2": Value::Array(vec![Value::Null, Value::from(2), Value::Null]),
+                    "key3": Value::from(2),
+                ],
+                CompactOptions {
+                    recursive: true,
+                    null: true,
+                    ..Default::default()
+                },
+            ),
+        ];
+
+        for (expected, original, options) in cases {
+            assert_eq!(expected, compact_map(original, &options))
+        }
+    }
+
+    #[test]
+    fn compact() {
+        let cases = vec![
+            (
+                map![
+                    "foo":
+                        map!["key1": Value::Null,
+                             "key2": 1,
+                             "key3": "",
+                        ]
+                ],
+                Ok(Value::Map(map!["key2": 1])),
+                CompactFn::new(
+                    Box::new(Path::from("foo")),
+                    Some(Box::new(Literal::from(true))),
```

Comment:
`Literal` has a `boxed()` method defined on it, to make these tests a bit easier on the eyes.